api/campaigns/class-maybe-show-campaign.php

```diff
@@ -74,15 +74,49 @@ function ( $post_data ) use ( $visit ) {
 		}
 
 		$page_referer_url = isset( $_REQUEST['ref'] ) ? $_REQUEST['ref'] : ''; // phpcs:ignore WordPress.Security.NonceVerification.Recommended, WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
+
+		$overlay_to_maybe_display      = null;
+		$above_header_to_maybe_display = null;
+
 		foreach ( $campaigns as $campaign ) {
-			$response[ $campaign->id ] = $this->should_campaign_be_shown(
+			$campaign_should_be_shown = $this->should_campaign_be_shown(
 				$client_id,
 				$campaign,
 				$settings,
 				filter_input( INPUT_SERVER, 'HTTP_REFERER', FILTER_SANITIZE_STRING ),
 				$page_referer_url,
 				$view_as_spec
 			);
+
+			// If an overlay is already able to be shown, pick the one that has the higher priority.
+			if ( $campaign_should_be_shown && 'o' === $campaign->t ) {
+				if ( empty( $overlay_to_maybe_display ) ) {
+					$overlay_to_maybe_display = $campaign;
+				} else {
+					$higher_priority = self::get_higher_priority( $overlay_to_maybe_display, $campaign, $settings->all_segments );
```

Comment:
A naming suggestion, to make it more understandable:
```suggestion
					$higher_priority_item = self::get_higher_priority_item( $overlay_to_maybe_display, $campaign, $settings->all_segments );
```

Reply:
Updated in 6a5f9ed.